filesystem/Upload.php

```diff
@@ -384,7 +405,22 @@ public function setAllowedMaxFileSize($rules) {
 		if(is_array($rules) && count($rules)) {
 			// make sure all extensions are lowercase
 			$rules = array_change_key_case($rules, CASE_LOWER);
-			$this->allowedMaxFileSize = $rules;
+			$finalRules = array();
+			$tmpSize = 0;
+			
+			foreach ($rules as $rule => $value) {
+				if (is_string($value)) {
```

Comment:
I'm worried this type check is too liberal (eg: '1024' could evaluate as true).

Can we do `if (is_numeric(...))` and swap the code round so that numeric strings are allowed?